cmd/system-probe/api/server/listener.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package server contains system-probe API server functionality
+package server
+
+import "net"
+
+// Listener is a wrapper over some net.Listener
+type Listener interface {
+	// GetListener returns the underlying net.Listener
+	GetListener() net.Listener
+
+	// Stop and clean up resources for the underlying connection
+	Stop()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why do you need the `Stop()` method since there is already the `Close()` method in `net.Listener` ?

[Golang net package takes care of unlinking the socket on close](https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/net/unixsock_posix.go;l=179-197)

Reply:
Updated to remove the custom `Listener` abstraction